ckan/lib/navl/dictization_functions.py

```diff
@@ -219,35 +220,22 @@ def augment_data(data, schema):
 
 
 def convert(converter, key, converted_data, errors, context):
-
-    try:
-        value = converter(converted_data.get(key))
-        converted_data[key] = value
-        return
-    except TypeError as e:
-        # hack to make sure the type error was caused by the wrong
-        # number of arguments given.
-        if converter.__name__ not in str(e):
-            raise
-    except Invalid as e:
-        errors[key].append(e.error)
-        return
-
-    try:
-        converter(key, converted_data, errors, context)
-        return
-    except Invalid as e:
-        errors[key].append(e.error)
-        return
-    except TypeError as e:
-        # hack to make sure the type error was caused by the wrong
-        # number of arguments given.
-        if converter.__name__ not in str(e):
-            raise
-
+    signature = inspect.signature(converter)
```

Comment:
does `inspect.signature` do a lot of work that we don't need to just get the number of arguments (i.e. `converter.__code__.co_argcount`)? We call this function many times when running validation, have you measured the performance impact of this change?

Reply:
I haven't used __code__ object because it's not available for built-ins. But apparently signature has the same limitations - it can't inspect functions like `str`. So I'll search for other options

Reply:
Hmm, it seems that my solution won't work because all the introspection tools work only win python-fuctions. Even though it may be rare and we can update all such cases in CKAN, in 3rd-party extension people still may want to use `str/list/dict` as validators, so we have to rely on the old way with TypeErrors till python will provide even-more stable option to inspect functions :(

Reply:
dunder names like `__code__` aren't internals, they're just special names designed not to conflict with anything a user might add.  I'm happy to prevent things like `str` from being used as a validator because they can raise exceptions we're not expecting.

Reply:
Cool, in that case i'll replace such functions with simple wrappers and switch to `__code__` for checks.

BTW, `__code__` x3 times faster than `try\except` version